internal/provider/kubernetes/kubernetes.go

```diff
@@ -40,6 +40,14 @@ func New(cfg *rest.Config, svr *config.Server, resources *message.ProviderResour
 		LeaderElectionID:       "5b9825d2.gateway.envoyproxy.io",
 		MetricsBindAddress:     ":8080",
 	}
+
+	if svr.EnvoyGateway.Provider != nil &&
+		svr.EnvoyGateway.Provider.Kubernetes != nil &&
+		(svr.EnvoyGateway.Provider.Kubernetes.Watch != nil) &&
+		(len(svr.EnvoyGateway.Provider.Kubernetes.Watch.Namespaces) > 0) {
+		mgrOpts.Cache.Namespaces = svr.EnvoyGateway.Provider.Kubernetes.Watch.Namespaces
```

Comment:
will EG only get resources from `WatchNamespaces`?

Reply:
yes if this field is set, the test case reflects this